prov/rxm/src/rxm_ep.c

```diff
@@ -512,13 +516,21 @@ static ssize_t rxm_ep_send_common(struct fid_ep *ep_fid, const struct iovec *iov
 		pkt_size = sizeof(*pkt) + pkt->hdr.size;
 	}
 
-	ret = fi_send(rxm_conn->msg_ep, pkt, pkt_size, desc_tx_buf, 0, tx_entry);
-	if (ret) {
-		FI_WARN(&rxm_prov, FI_LOG_EP_DATA, "fi_send for MSG provider failed\n");
-		goto err;
+	if (flags & FI_INJECT & ~FI_COMPLETION) {
+		ret = fi_inject(rxm_conn->msg_ep, pkt, pkt_size, 0);
```

Comment:
My suggestion  is to use fi_sendmsg instead of fi_send and fi_inject. I meant that provider on which RXM based will decide how to data should sent. But right now I guess it is not good idea :) I will add verification for pkt_size (if pkt_size > MSG will return -FI_ENODATA error). 

I've tested RXM based on new provider in fabtests (when new provider will be ready for release, will send these changes - we think it will be the end of March) and I see that FI_SELECTIVE_COMPLETION works well for RXM. I.e. I can confirm that all cases which are described for this flag in "fi_endpoint" in man pages are workable for RXM.